src/release-notes.ts

```diff
@@ -0,0 +1,150 @@
+import * as core from '@actions/core'
+import * as common from './common'
+import { GitHub, Context } from './types'
+import * as cheerio from 'cheerio'
+
+const pendingDecisionLabel = 'pending:release-notes'
+const hasNotesLabel = 'has:release-notes'
+const notReleaseNoteWorthyLabel = 'has:release-notes-decision'
+
+const worthyLabels = new Set<string>(['a:feature', 'a:regression', 'a:performance-improvement', 'a:epic'])
+
+//eslint-disable-next-line prettier/prettier
+const releaseNotesPaths = new Set<string>([
+  'subprojects/docs/src/docs/release/notes.md',
+  'platforms/documentation/docs/src/docs/release/notes.md'
+])
+
+function shouldHaveReleaseNotes(issue: any): boolean {
+  const labels = issue.labels.nodes.map((label: any) => label.name)
+
+  //eslint-disable-next-line prettier/prettier
+  return labels.some((label: string) => worthyLabels.has(label)) &&
+    !labels.includes(notReleaseNoteWorthyLabel) &&
+    !labels.includes(hasNotesLabel) //in case notes are updated in an unrelated PR
+}
+
+// see https://github.com/orgs/community/discussions/24492
+// and https://github.com/orgs/community/discussions/24367#discussioncomment-3243930
+async function getLinkedPrNumbers(context: Context, issue: any): Promise<number[]> {
+  const response = await fetch(`https://github.com/${context.repo.owner}/${context.repo.repo}/issues/${issue.number}`)
+  const html = await response.text()
+  const document = cheerio.load(html)
+  const prNumbers = Array.from(document('development-menu form span a'))
```

Comment:
😨 Yikes, that sounds ... brittle ... but I guess there is no better way for now 😢

Reply:
Exactly :harold: